src/corona_hakab_model/update_matrix.py

```diff
@@ -1,8 +1,28 @@
 from generation.connection_types import ConnectionTypes
+from generation.circles import SocialCircle
 from agent import Agent
 import numpy as np
-from typing import Iterable
+from typing import Iterable, Callable, Any
 
+class Policy:
```

Comment:
In total looks good and generic,
We need to make sure this also goes for all kinds of policies applied, for example -
- We send only people marked "detected" home, making them unconnected to all society, except family.
- Make some changes only for certain ages of the population

etc...